pipeline/objects/decorators.py

```diff
@@ -54,7 +51,13 @@ def execute_func(*args, **kwargs):
             return node_output
 
     execute_func.__function__ = function
+
+    function.__run_once__ = run_once
```

Comment:
I think double underscores before and after a variable name are typically used for Python built-in magic methods (https://towardsdatascience.com/whats-the-meaning-of-single-and-double-underscores-in-python-3d27d57d6bd1).

Not sure if there's a reason we're using them specifically here rather than just something like `function._run_once`, but I see we're already doing it in a few other places, so I guess it's consistent with the current conventions we're using?

Reply:
Yeah you're right, I'm not sure what the original motivation was either. I did this to keep the consistency.